setup.pp

```diff
@@ -96,13 +96,20 @@
   'pygame',
   'pyserial',
   'requests',
+  'cherrypy',
 ]
 
 python::pip { $python_packages:
   pip_provider => 'pip3',
   require      => Package[ $main_packages, $piweatherrock_packages, ],
 }
 
+# Run upgrade script to import current config values to new config file
+exec { 'import config':
+  path    => '/bin:/usr/bin',
+  command => "python3 /home/pi/PiWeatherRock/scripts/upgrade.py",
+}
```

Comment:
As this stands, it will run every time puppet runs. For most this would be fine but for me this would be a problem as I run puppet from cron to keep things in check. Maybe look at how you could incorporate an `unless` attribute here. https://puppet.com/docs/puppet/latest/types/exec.html#exec-attribute-unless

Reply:
Good catch. I think this is solved. I check if `config.json-sample` exists. The upgrade will run if it does. After the upgrade completes, `config.json-sample` is either deleted or has been updated and renamed to `config.json`. This should prevent the script from executing, except when puppet is executed for the first time directly after upgrading the application.

Reply:
Removing the file will only work if it’s in the .gitignore which doesn’t seem practical. I think you may want to key off of the non-sample version that is ignored.

Reply:
Oh, didn't know about the issue with deleting tracked files. That's good to know. 
I've set it to `grep` for the version key in the config, and key off of that value instead.